pkg/authorization/authorizer/authorizer.go

```diff
@@ -394,6 +436,26 @@ func (a *openshiftAuthorizationAttributeBuilder) GetAttributes(req *http.Request
 		return nil, errors.New("could not get user")
 	}
 
+	isNonResourceURL := a.infoResolver.APIPrefixes.Has(req.URL.Path[1:])
```

Comment:
> Can you invert the naming here?

Inverted for the test, but left the backwards seeming name on the attributes.  That way, the default object is left with IsNonResource==false.  That lets the common case where we're touching resources only touch the fields it expects to need.

If you feel strongly (and I wouldn't leave the backwards name except for the default object), I can switch it.